python/paddle/fluid/backward.py

```diff
@@ -255,6 +261,9 @@ def _add_needed_descs_to_block(descs, block, main_block, in_memory_vars):
             if name not in in_memory_vars:
                 is_needed = True
         if is_needed:
+            if origin_is_operator:
+                global _grad_op_id_to_fwd_op
+                _grad_op_id_to_fwd_op[desc.original_id()] = origin_desc
```

Comment:
What if `origin_is_operator` is false?

Reply:
It means that the desc is an instance of OpDesc on cpp side, it has been added to the map earlier when creating that cpp desc. Here we only handle if desc is on instance of Operation on python side, which is used for recompute's ff_ops.